api/suggested-fees.js

```
@@ -0,0 +1,68 @@
+// Note: ideally this would be written in ts as vercel claims they support it natively.
+// However, when written in ts, the imports seem to fail, so this is in js for now.
+
+
+const sdk = require("@uma/sdk-next");
+const { BridgeAdminEthers__factory } = require("@uma/contracts-node");
+const ethers = require("ethers");
+
+const handler = async (request, response) => {
+  try {
+    const { REACT_APP_PUBLIC_INFURA_ID } = process.env;
+    const provider = new ethers.providers.JsonRpcProvider(`https://mainnet.infura.io/v3/${REACT_APP_PUBLIC_INFURA_ID}`);
+
+    let { amount, l2Token, chainId } = request.query;
+    if (!isString(amount) || !isString(l2Token) || !isString(chainId)) throw new InputError("Must provide amount and token as query params");
+
+    let { l1Token } = await getTokenDetails(provider, l2Token, chainId);
+    if (l1Token === sdk.across.constants.ADDRESSES.WETH) l1Token = sdk.across.constants.ADDRESSES.WETH;
+    const depositFeeDetails = await sdk.across.gasFeeCalculator.getDepositFeesDetails(provider, amount, l1Token === sdk.across.constants.ADDRESSES.WETH ? sdk.across.constants.ADDRESSES.ETH : l1Token);
```

do u want to throw an error if the amount isnt high enough?

---

I was planning on _only_ erroring if the contract would reject it, and I was thinking based on our conversation last week that the SDK would do that. Do you think that belongs here or in the SDK?

---

the sdk already does this calculation but i decided to return a flag rather than throw an error. You can just check the flag and throw an error directly: `isAmountTooLow`

---

oh btw the sdk does not call the contract directly for this check, so maybe its not the same as what u are asking for.

---

Done.